src/BugsnagUnityForReview.mm

```diff
@@ -0,0 +1,876 @@
+
+    void bugsnag_setBreadcrumbMetadata(const void *breadcrumb, const char *jsonString);
+
+    const char * bugsnag_getBreadcrumbMetadata(const void *breadcrumb);
+
+    void bugsnag_getBreadcrumbsFromEvent(const void *event,const void *instance, void (*callback)(const void *instance,void *breadcrumbs[], int breadcrumbs_size));
+
+    void bugsnag_getErrorsFromEvent(const void *event,const void *instance, void (*callback)(const void *instance, void *errors[], int errors_size));
+
+    void bugsnag_getStackframesFromError(const void *error,const void *instance, void (*callback)(const void *instance, void *stackframes[], int stackframes_size));
+
+    void bugsnag_getStackframesFromThread(const void *thread,const void *instance, void (*callback)(const void *instance, void *stackframes[], int stackframes_size));
+
+    const char * bugsnag_getSeverityFromEvent(const void *event);
+
+    void bugsnag_setEventSeverity(const void *event, const char *severity);
+
+    void bugsnag_getThreadsFromEvent(const void *event,const void *instance, void (*callback)(const void *instance, void *threads[], int threads_size));
+
+    BugsnagUser * bugsnag_getUserFromEvent(const void *event);
+
+    BugsnagUser * bugsnag_getUserFromSession(const void *session);
+
+    void bugsnag_setEventMetadata(const void *event, const char *tab, const char *metadataJson);
+
+    void bugsnag_clearEventMetadataSection(const void *event, const char *section);
+
+    void bugsnag_clearEventMetadataWithKey(const void *event, const char *section, const char *key);
+
+    const char * bugsnag_getEventMetaData(const void *event, const char *tab);
+
+    const char * bugsnag_getErrorTypeFromError(const void *error);
+
+    const char * bugsnag_getThreadTypeFromThread(const void *thread);
+
+    const char * bugsnag_getValueAsString(const void *object, char *key);
+
+    void bugsnag_setNumberValue(const void *object, char * key, const char * value);
+
+
+}
+
+const char * getMetadataJson(NSDictionary* dictionary){
+
+    if (!dictionary) {
+        return NULL;
+    }
+
+    @try {
+        NSError *error = nil;
+        NSData *data = [NSJSONSerialization dataWithJSONObject:dictionary options:0 error:&error];
+        if (data) {
+            return strndup((const char *)data.bytes, data.length);
+        } else {
+            NSLog(@"%@", error);
+        }
+    } @catch (NSException *exception) {
+        NSLog(@"%@", exception);
+    }
+    return NULL;
+}
+
+NSDictionary * getDictionaryFromMetadataJson(const char * jsonString){
+    @try {
+        NSError *error = nil;
+        NSData *data = [NSData dataWithBytesNoCopy:(void *)jsonString length:strlen(jsonString) freeWhenDone:NO];
+        id metadata = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+        if ([metadata isKindOfClass:[NSDictionary class]]) {
+            return metadata;
+        }
+    } @catch (NSException *exception) {
+        NSLog(@"%@", exception);
+        return nil;
+    }
+}
```

Comment:
This function is missing a return value in the case that no exception is thrown but metadata is not a dictionary.

```suggestion
    } @catch (NSException *exception) {
        NSLog(@"%@", exception);
    }
    return nil;
}
```

Reply:
Done 👍